packages/loaders-compressed-textures/src/CompressedTextureLoader.ts

```diff
@@ -0,0 +1,188 @@
+     *   fallback: "asset.png"
+     * });
+     * ```
+     *
+     * @param resource
+     * @param next
+     */
+    static use(resource: ILoaderResource, next: (...args: any[]) => void): void
+    {
+        const data: CompressedTextureManifest = resource.data;
+        const loader = (this as any) as Loader;
+
+        if (!CompressedTextureLoader.textureExtensions)
+        {
+            CompressedTextureLoader.autoDetectExtensions();
+        }
+        if (resource.type === LoaderResource.TYPE.JSON
+            && data
+            && data.cacheID
+            && data.textures)
+        {
+            const textures = data.textures;
+
+            let textureURL: string;
+            let fallbackURL: string;
+
+            // Search for an extension that holds one the formats
+            for (let i = 0, j = textures.length; i < j; i++)
+            {
+                const texture = textures[i];
+                const url = texture.src;
+                const format = texture.format;
+
+                if (!format)
+                {
+                    fallbackURL = url;
+                }
+                if (CompressedTextureLoader.textureFormats[format])
+                {
+                    textureURL = url;
+                    break;
+                }
+            }
+
+            textureURL = textureURL || fallbackURL;
+
+            // Make sure we have a URL
+            if (!textureURL)
+            {
+                throw new Error(`Cannot load compressed-textures in ${resource.url}, make sure you provide a fallback`);
+            }
+            if (textureURL === resource.url)
+            {
+                // Prevent infinite loops
+                throw new Error('URL of compressed texture cannot be the same as the manifest\'s URL');
+            }
+
+            const loadOptions = {
+                crossOrigin: resource.crossOrigin,
+                metadata: resource.metadata.imageMetadata,
+                parentResource: resource
+            };
+
+            // The appropriate loader should register the texture
+            loader.add(data.cacheID, textureURL, loadOptions, function onCompressedTextureLoaded()
+            {
+                next();
+            });
+        }
+        else
+        {
+            next();
+        }
+    }
+
+    /**
+     * Detects the available compressed texture extensions on the device.
+     *
+     * @param extensions - extensions provided by a WebGL context
+     * @ignore
+     */
+    static autoDetectExtensions(extensions?: Partial<CompressedTextureExtensions>): void
+    {
+        // Auto-detect WebGL compressed-texture extensions
+        if (!extensions)
+        {
+            const canvas = document.createElement('canvas');
+            const gl = canvas.getContext('webgl');
+
+            if (!gl)
+            {
+                console.error('WebGL not available for compressed textures. Silently failing.');
+
+                return;
+            }
+
```

Comment:
one for later, but feel like managing of all extensions might benefit from being centralised?

Reply:
Yes.